offchain/dusd-api/src/Dusd/Api.purs

```diff
@@ -0,0 +1,145 @@
+module Dusd.Api
+  ( initProtocolSimple
+  , updateProtocl
+  -- Types
+  , Protocol(..)
+  -- Testing
+  , getWalletPubkeyhash
+  , mintNft
+  , seedTx
+  ) where
+
+import Contract.Prelude
+
+import Contract.Address (PubKeyHash, getWalletAddress, getWalletCollateral, scriptHashAddress)
+import Contract.Credential (Credential(..))
+import Contract.Log (logDebug', logInfo')
+import Contract.Monad (Contract, liftContractM)
+import Contract.PlutusData (Datum(..), OutputDatum(..), PlutusData(..), Redeemer(..), toData)
+import Contract.ScriptLookups as Lookups
+import Contract.Scripts (Validator, validatorHash)
+import Contract.Transaction (TransactionInput, TransactionOutput(..), TransactionOutputWithRefScript(..))
+import Contract.TxConstraints (DatumPresence(..), TxConstraints)
+import Contract.TxConstraints as Constraints
+import Contract.Utxos (getUtxo)
+import Contract.Value (CurrencySymbol, adaToken, scriptCurrencySymbol)
+import Contract.Value as Value
+import Ctl.Internal.Plutus.Types.Address (Address(..))
+import Ctl.Utils (buildBalanceSignAndSubmitTx, getUtxos, waitForTx)
+import Data.Array (cons)
+import Data.BigInt as BigInt
+import Data.List (head)
+import Data.Map (keys, singleton)
+import Data.Map as Map
+import Data.Set (toUnfoldable)
+import Dusd.CborTyped (configAddressValidator, simpleNft)
+import Effect.Exception (throw)
+
+newtype Protocol = Protocol { datum :: PlutusData, utxo :: TransactionInput, nftCs :: CurrencySymbol, configVal :: Validator }
```

Comment:
Can we have the aeson instances here already?

Reply:
Sure, I'll set that up.